wallet-sdk/src/main/kotlin/org/stellar/walletsdk/Wallet.kt

```diff
@@ -96,22 +124,49 @@ internal data class Config(val stellar: StellarConfiguration, val app: Applicati
  * example, `android.util.Base64`.
  * @property useHttp when enabled, switch from https to http scheme. Only allowed when network is
  * not [Network.PUBLIC] for security reasons
+ * @property defaultClientConfig configuration for default client used across the app.
  */
 data class ApplicationConfiguration(
   val defaultSigner: WalletSigner = WalletSigner.DefaultSigner(),
   val base64Decoder: Base64Decoder = defaultBase64Decoder,
-  val useHttp: Boolean = false
-)
+  val defaultClientConfig: ClientConfigFn = {}
+) {
+  @Suppress("MaxLineLength")
+  @Deprecated(
+    "Can be configured using defaultClientConfig",
+    replaceWith =
+      ReplaceWith(
+        "ApplicationConfiguration(defaultSigner, base64Decoder) { defaultRequest { url { protocol = URLProtocol.HTTP } } }",
+        "io.ktor.client.plugins.defaultRequest",
+        "io.ktor.http.URLProtocol"
+      )
+  )
+  constructor(
+    defaultSigner: WalletSigner = WalletSigner.DefaultSigner(),
+    base64Decoder: Base64Decoder = defaultBase64Decoder,
+    useHttp: Boolean
+  ) : this(
+    defaultSigner,
+    base64Decoder,
+    {
+      if (useHttp) {
+        defaultRequest { url { protocol = URLProtocol.HTTP } }
+      }
+    }
+  )
+
+  val defaultClient =
+    HttpClient(OkHttp) {
+      install(ContentNegotiation) { json(defaultJson) }
+      defaultRequest { url { protocol = URLProtocol.HTTPS } }
+      defaultClientConfig()
+    }
```

Comment:
It also might make sense to set reasonable defaults for things like the connect timeout. Check out the Python Stellar SDK [client](https://stellar-sdk.readthedocs.io/en/stable/_modules/stellar_sdk/client/requests_client.html#RequestsClient) and its defaults.

Reply:
We use the defaults from okhttp client